pkg/lib/authn/authenticator/password/checker.go

```diff
@@ -250,6 +260,23 @@ func (pc *Checker) checkPasswordHistory(password, authID string) (*Policy, error
 	return nil, nil
 }
 
+func (pc *Checker) checkPasswordExpiry(password, authID string) error {
```

Comment:
This checking should be done in password.Provider#Authenticate, rather than here.

Rationale
- password.Provider#Authenticate has the *authenticator.Password already.
- password.Checker is intended to enforce password requirement, not password expiry.